### ISSUE TITLE
Duplicates in product/model import result in SQL error

### ISSUE THREAD

**Issue by joshzamor** [akeneo/pim-community-dev#12899]
:bug: **I'm reporting a Bug** :bug:

## Bug

When importing a product or product model, if the file has a duplicate, a SQL error will be presented back to the user and the process will crash.

## Effected Versions

* Akeneo CE v.3.2.70
* Akeneo CE v4.0.56

## Steps to Reproduce

1. Create a product or product model import that has a duplicate.

    e.g. product model csv

    ```
    code,family,name
    123,EXAMPLE_FAMILY,some product model name
    123,EXAMPLE_FAMILY,some OTHER product model name
    ```
1. Import product / product model using the CSV or XLSX job as appropriate

## Current behavior

An error is displayed such as:

```
An exception occurred while executing 'INSERT INTO pim_catalog_product_model (code, raw_values, created, updated, root, lvl, lft, rgt, parent_id, family_variant_id) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)' with params [...]: SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '123' for key 'UNIQ_5943911E77153098' 
```

and

```
The EntityManager is closed
```

## Expected behavior

The error displayed to the users informs them that there is a duplicate entry in their import file.

**More ideal**:  The duplicate results in a warning to the user, but the import proceeds along, importing the duplicate and over-writing the details from the previous entry in the import file.

## Screenshots

![akeneo_demo_dupe_error](https://user-images.githubusercontent.com/6156389/94740322-53d4e080-0327-11eb-8b22-5a51d1b55026.png)

## Version info

From demo.akeneo.com: 
[akeneo-pim-system-info_2020-09-30_21 16.txt](https://github.com/akeneo/pim-community-dev/files/5308275/akeneo-pim-system-info_2020-09-30_21.16.txt)

## Relevant issues

#4298

**Comment by tseho** [comment]
Thanks for the report, I found the issue:

In `UniqueProductEntityValidator`, we first check in an in-memory set where currently created Products Identifiers are stored, and then in the database.
https://github.com/akeneo/pim-community-dev/blob/4cbfb24459af0d9ed6a0932abce9106268213513/src/Akeneo/Pim/Enrichment/Component/Product/Validator/Constraints/Product/UniqueProductEntityValidator.php#L56-L78

BUT in `UniqueProductModelEntityValidator`, for Product Models, the check is only done on the database:
https://github.com/akeneo/pim-community-dev/blob/4cbfb24459af0d9ed6a0932abce9106268213513/src/Akeneo/Pim/Enrichment/Component/Product/Validator/Constraints/Product/UniqueProductModelEntityValidator.php

And we cannot simply add `UniqueValuesSet` to it because it only support values of type `ValueInterface`, which a Product Model Code is not.

I will try to think of a good solution.

**Comment by tseho** [comment]
Hello @joshzamor,
The fix is now merged on master.
Which version are you using ?

**Comment by joshzamor** [comment]
That's great news, thanks @tseho .

I'm currently on 3.2.70, thought it'd be easy for me to upgrade to the latest patch release in the 3.2.x line.  Is that possible?

**Comment by Schwierig** [comment]
@tseho When or how will this bug fix be merged into the current version branches?

**Comment by tseho** [comment]
Hello both of you,
Sorry for the delay, I've backported the fix to 3.2 in https://github.com/akeneo/pim-community-dev/pull/13009

**Comment by tseho** [comment]
It's merged and will be included in 3.2.75 when released.
If that's good, @joshzamor, I will let you close the issue.
Have a great day.

**Comment by joshzamor** [comment]
Thank you @tseho and @Schwierig , that works great, thank you.
